testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/mdb/dynamic/adapter/TelnetResourceAdapter.java

```diff
@@ -84,10 +84,13 @@ public void run() {
 
     @Override
     public void endpointDeactivation(MessageEndpointFactory messageEndpointFactory, ActivationSpec activationSpec) {
-        final TelnetActivationSpec telnetActivationSpec = (TelnetActivationSpec) activationSpec;
-
         final TelnetServer telnetServer = activated.remove(port);
 
+        // Though this is not likely to happen, we'd better check it and throw the NullPointerException with controlled message.
+        if (telnetServer == null) {
+            throw new NullPointerException(String.format("telnetServer for port %s is null!", port));
+        }
```

Comment:
No need to change this, but below you could use:
```java
Objects.requireNonNull(String.format("No telnetServer found for port %d", port), teletServer).deactivate();
```

Reply:
@jamezp Thanks for teaching!